src/memjs/memjs.ts

```diff
@@ -384,6 +399,48 @@ class Client<Value = MaybeBuffer, Extras = MaybeBuffer> {
     return Object.assign({}, ...results);
   }
 
+  async getMultiWithErrors<Keys extends string>(
+    keys: Keys[]
+  ): Promise<GetMultiWithErrorsResult<Keys, Value, Extras>> {
+    const serverKeytoLookupKeys: {
+      [serverKey: string]: Keys[];
+    } = {};
+    keys.forEach((lookupKey) => {
+      const serverKey = this.lookupKeyToServerKey(lookupKey);
+      if (!serverKeytoLookupKeys[serverKey]) {
+        serverKeytoLookupKeys[serverKey] = [];
+      }
+      serverKeytoLookupKeys[serverKey].push(lookupKey);
+    });
+
+    const usedServerKeys = Object.keys(serverKeytoLookupKeys);
+    const errors: GetMultiError<Keys>[] = [];
+    const results = await Promise.all(
```

Comment:
in case of error, will results contains undefined for the keys or missing?

Reply:
results will be an array of partial results and undefineds, so if we had 3 keys foo, bar, and baz on different servers and the server for bar failed, we'd get:  
```
[{foo: {value: "1", extras: "", cas: "token"}},
  undefined,
  {baz: {value: "2", extras: "", cas: "token"}]
```

This would be unpacked/coalesced by `Object.assign({}, ...results)` to yield

```
{
  foo: {value: "1", extras: "", cas: "token"},
  baz: {value: "2", extras: "", cas: "token"}
}
```

from looking at the code/types, I think if bar were a cache miss instead of a failure we'd get

```
[{foo: {value: "1", extras: "", cas: "token"}},
  {},
  {baz: {value: "2", extras: "", cas: "token"}]
```

which would yield the same result

but I would need to verify and test